pkg/controllers/crds/crds_controller.go

```diff
@@ -240,14 +248,18 @@ func injectService(crd *apiext.CustomResourceDefinition, svc types.NamespacedNam
 	return nil
 }
 
-func injectCert(crd *apiext.CustomResourceDefinition, certPem []byte) error {
+func injectCert(crd *apiext.CustomResourceDefinition, certPem []byte) (bool, error) {
```

Comment:
returning the `bool` here just for the unit tests has a smell. Please change the testing approach.

Instead of having sequential tests, where each test depends on the previous one, try to aim for table driven tests and specify the input and expected output of the `injectCert` func. Most IDEs are capable of generating table driven tests for a function.